### ISSUE TITLE
Action creation is timing out

### ISSUE THREAD

**Issue by ctrombley** [auth0/terraform-provider-auth0#103]
#### Summary
Our TF apply is timing out without any useful information due to the management API returning `"status": "failed"` in a response from `POST api/v2/actions/actions`.  Here is the code responsible for the timeout: 

https://github.com/auth0/terraform-provider-auth0/blob/main/auth0/resource_auth0_action.go#L202

The `go-auth0` client method `Action.Create()` [is not returning an error](https://github.com/auth0/go-auth0/blob/main/management/management.go#L299-L323) when the API returns `"status": "failed"`, so this is not caught as a `NonRetryableError`.  If `deploy` is set to `true`, during `deployAction` the [provider just retries](https://github.com/auth0/terraform-provider-auth0/blob/main/auth0/resource_auth0_action.go#L196-L209) until `TimeoutCreate` is reached, eventually spitting out the unhelpful message `Expected action status "failed" to equal "built"`.

The problem here was ultimately due to errors in the action config, but it took us a very long time to figure that out.  

I have some suggestions and I'd like to put up a PR to resolve.  This issue blocked internal teams here at Indigo for a while and has taken up a lot of our resources already.

1. Is it possible for the management API to return more information about *why* the build failed? Once I'm in the Auth0 UI I can see lots of suggestions in the template editor.  Can those be sent back in the API response?  If not, something like "Please check your resource in the UI" as a hint about what to do next would be way more helpful than `Expected action status "failed" to equal "built"`.
2. There should be some code that fails fast when `"status": "failed"` is sent back from the API.  Should that go in [go-auth0/management](https://github.com/auth0/go-auth0/blob/main/management/management.go#L299-L323)?  Should it be an api-wide check?  Or should we handle it locally [in the provider](https://github.com/auth0/terraform-provider-auth0/blob/main/auth0/resource_auth0_action.go#L196-L209)? Again, I'd like to make a PR so implementation advice would be appreciated.

**Comment by sergiughf** [comment]
Hey @ctrombley, thanks for reporting this issue to us. Appreciate the detailed summary of the issue and apologies for the inconvenience. Our team will take a look at this and patch it soon. Leaving the issue open until then.

**Comment by ctrombley** [comment]
@sergiughf I put up a PR to resolve the issue.

If you take a look at the acceptance test I wrote, one of the main problems we ran into was that the following dependency definition was causing the build to fail silently, without any indication in the UI of what went wrong:

```
runtime = "node16"
dependencies {
    name    = "crypto"
    version = "17.7.1"
}
```

My guess is that it's because [crypto has been deprecated](https://www.npmjs.com/package/crypto) and is now part of node core, and there was some dependency resolution conflict?  It was a process of trial & error and removing sections from the config until we were able to find out what was wrong with the action config.    This is a problem that extends to the UI and management API, and I hope you're able to pass it along to teams that can put some better feedback in place for users that are facing the same problem as we did.
